tools/Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM python:3.10-slim-bullseye
+MAINTAINER Opentrons Engineering <engineering@opentrons.com>
+
+ENV BUILDENV_ROOT=/build-environment
+ENV POETRY_HOME=/build-environment/poetry
+ENV POETRY_VERSION=1.2.1
+ENV PATH=$PATH:/build-environment/poetry/bin
+ENV POETRY_VIRTUALENVS_IN_PROJECT=true
+ENV HOME=/build-environment
+
+RUN apt update && apt install -y curl wget file git
+RUN rm /bin/sh && ln -s /bin/bash /bin/sh
+RUN mkdir /build-environment
+ADD . /build-environment/tools
+
+# TODO: This should be a release version of the SDK, but one does not
+# exist yet.
+RUN \
+    cd /build-environment \
+    && wget https://opentrons-buildroot-ci.s3.amazonaws.com/439bccec-fff5-4501-9074-f463408da59a/opentrons-buildroot/arm-buildroot-linux-gnueabihf_sdk-buildroot.tar.gz \
+    && tar xf ./arm-buildroot-linux-gnueabihf_sdk-buildroot.tar.gz \
+    && cd arm-buildroot-linux-gnueabihf_sdk-buildroot \
+    && ./relocate-sdk.sh
+
+RUN mkdir -p /${POETRY_HOME}/poetry \
```

Comment:
To increase cache hit ratio, you can break up this command into

1. Downloading and configuring tools (won't change that often unless configuration or poetry version changes)
2. Installing dependencies: `poetry install`
3. Running a build: `poetry build`
4. Everything else